src/controllers/dynakube/activegate/internal/proxy/reconciler_test.go

```diff
@@ -0,0 +1,177 @@
+	"context"
+	"testing"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/capability"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/consts"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	testDynakubeName       = "test-dynakube"
+	testNamespace          = "test-namespace"
+	customProxySecret      = "testProxy"
+	proxyUsername          = "testUser"
+	proxyPassword          = "secretValue"
+	proxyPort              = "1020"
+	proxyHost              = "proxyserver.net"
+	proxyDifferentUsername = "differentUsername"
+)
+
+func newTestReconciler(client client.Client) *Reconciler {
+	instance := &dynatracev1beta1.DynaKube{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: testNamespace,
+			Name:      testDynakubeName,
+		},
+		Spec: dynatracev1beta1.DynaKubeSpec{
+			APIURL: "https://testing.dev.dynatracelabs.com/api",
+		},
+	}
+
+	r := NewReconciler(client, client, instance)
+	return r
+}
+
+func TestReconcileWithoutProxy(t *testing.T) {
+	t.Run(`reconcile dynakube without proxy`, func(t *testing.T) {
+		r := newTestReconciler(fake.NewClientBuilder().Build())
+		update, err := r.Reconcile()
+
+		var proxySecret corev1.Secret
+		var clientError = r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)
+
+		assert.Empty(t, proxySecret)
+		assert.True(t, update)
+		assert.True(t, k8serrors.IsNotFound(clientError))
+		assert.NoError(t, err)
+	})
+	t.Run(`ensure proxy secret deleted`, func(t *testing.T) {
+		var testClient = fake.NewClientBuilder().WithObjects(&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      capability.BuildProxySecretName(),
+				Namespace: testNamespace,
+			},
+		}).Build()
+		r := newTestReconciler(testClient)
+		update, err := r.Reconcile()
+
+		var proxySecret corev1.Secret
+		var clientError = r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)
+
+		assert.Empty(t, proxySecret)
+		assert.True(t, update)
+		assert.True(t, k8serrors.IsNotFound(clientError))
+		assert.NoError(t, err)
+	})
+}
+
+func TestReconcileProxyValue(t *testing.T) {
+	t.Run(`reconcile proxy Value`, func(t *testing.T) {
+		var proxyValue = buildProxyUrl(proxyUsername, proxyPassword, proxyHost, proxyPort)
+		r := newTestReconciler(fake.NewClientBuilder().Build())
+		r.dynakube.Spec.Proxy = &dynatracev1beta1.DynaKubeProxy{Value: proxyValue}
+		update, err := r.Reconcile()
+
+		var proxySecret corev1.Secret
+		r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)
+
```

Comment:
the error shoudn't be implicitly ignored here
either check it
```suggestion
		update, err := r.Reconcile()

        assert.NoError(t, err)

		var proxySecret corev1.Secret
		err = r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)

```
or explicitly ignore it
```suggestion
		update, err := r.Reconcile()

		var proxySecret corev1.Secret
		_ = r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)

```